### DISCUSSION TITLE
Adding a custom DimReduction for the projection of cells into a pre-existing low dimensional space

### DISCUSSION THREAD

**Discussion by RegnerM2015** [GreenleafLab/ArchR#1147] [Feature Requests]
Hi @rcorces and @jgranja24 

I am trying to figure out if I can replicate the approach taken in this project (https://github.com/GreenleafLab/MPAL-Single-Cell-2019). In this work, a normal reference is created using only healthy cells. Then, disease cells are projected into the same low-dimensional space using the variable features of the normal reference. This framework is very different from just clustering/processing healthy and disease cells together as this explores a different and wider range of variable features.

Is it possible to implement this framework in the ArchR workflow? I think it would require a function to add a custom DimReduction similar to the one mentioned here: https://github.com/GreenleafLab/ArchR/discussions/572

I found the `.projectLSI` function from the source code which could be used to perform such a task. https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/IterativeLSI.R#L1234

Thank you!

**Comment by rcorces** [comment]
I agree that this is a useful feature to add. We dont have the hands to implement new features at the moment so this isnt on the horizon but we will take it into account as things move forward.

**Reply by RegnerM2015** [reply]
Thanks! I may be able to implement via pull request, but I will keep you updated!

**Reply by rcorces** [reply]
would be very happy to review a pull request!
